brightStars_DM_sprint_Feb2020/BeyondRadius.ipynb

```diff
@@ -0,0 +1,422 @@
+    "from lsst.daf.persistence import Butler\n",
+    "from lsst import geom\n",
+    "import matplotlib.pyplot as plt\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And script with utility functions:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import helperFuncs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Token to avoid rereading the detections every time, eg if we want to play with annuli parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "src_loaded = False "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Initialize Butler repo:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "butler = Butler(butlerPath)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Path to masks. Note there's likely some that are repeated when using Arcturus near the edges of patches."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "r\n"
+     ]
+    }
+   ],
+   "source": [
+    "if gouldingMask:\n",
+    "    # filenames have lower case filter in them:\n",
+    "    if chosenFilter=='HSC-G':\n",
+    "        filter_lc = 'g'\n",
+    "    elif chosenFilter=='HSC-R':\n",
+    "        filter_lc = 'r'\n",
+    "    elif chosenFilter=='HSC-I':\n",
+    "        filter_lc = 'i'\n",
+    "    elif chosenFilter=='HSC-z':\n",
+    "        filter_lc = 'z'\n",
+    "    elif chosenFilter=='HSC-y':\n",
+    "        filter_lc = 'y'\n",
+    "    print(filter_lc)\n",
+    "    mask_path = \"/datasets/hsc/BrightObjectMasks/GouldingMasksS18A/new_S18Amask_{}.reg\".format(\n",
+    "                                            filter_lc)\n",
+    "else:\n",
+    "    mask_path = \"/datasets/hsc/BrightObjectMasks/ArcturusMasks/\" "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If plots are to be saved, make sure the folder exists:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if savePlots:\n",
+    "    if not os.path.isdir('plots/'):\n",
+    "        os.makedirs('plots/')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run\n",
+    "Warning: running cell below can take a while."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "densities, brightMags, brightRads, brightCenters = [], [], [], []\n",
+    "for tract,patches in zip(tractList,patchList):\n",
```

Comment:
Does this work the way you think it does if your tractList contains more than one entry?  E.g.
```
In [47]: patchList = [["{},{}".format(i,j) for i in range(2,7) for j in range(2,7)]]                          

In [48]: tractList = [9813, 9697]                                                                             

In [49]: for tract, patches in zip(tractList,patchList): 
    ...:     print(tract, patches) 
    ...:                                                                                                      
9813 ['2,2', '2,3', '2,4', '2,5', '2,6', '3,2', '3,3', '3,4', '3,5', '3,6', '4,2', '4,3', '4,4', '4,5', '4,6', '5,2', '5,3', '5,4', '5,5', '5,6', '6,2', '6,3', '6,4', '6,5', '6,6']
```
(i.e. 9697 doesn't get looped over...you would need a duplicated "list of patches" in the patchList for each additional tract...)

It might be clearer just to split out the tract loop from that of the patches?

Reply:
This is actually vestigial from when I was using several tracts. I agree it would be clearer and cleaner to separate the two, but I like the idea that we can specify different patches for each tract we consider, so I went with

`["{},{}".format(i,j) for i in range(2,7) for j in range(2,7)] * len(tractList)`

instead - which should work the same way, without requiring any changes, for several tracts.